network/simple-onchain-discovery/src/lib.rs

```diff
@@ -96,9 +101,22 @@ fn extract_updates(role: RoleType, node_set: ValidatorSet) -> Vec<ConnectivityRe
 
     // Collect the set of EligibleNodes
     updates.push(ConnectivityRequest::UpdateEligibleNodes(
+        DiscoverySource::OnChain,
         node_list
             .iter()
-            .map(|node| (*node.account_address(), public_key(role, node.config())))
+            .map(|node| {
+                // TODO(philiphayes): remove this after removing pubkey field
+                let pubkey = public_key(role, node.config());
+                // parse out pubkey from address
+                let pubkey_set: HashSet<_> = network_address(role, node.config())
+                    .ok()
+                    .iter()
+                    .filter_map(NetworkAddress::find_noise_proto)
+                    // for now, also add the explicit pubkey field
+                    .chain(iter::once(pubkey))
+                    .collect();
+                (*node.account_address(), pubkey_set)
```

Comment:
what's the value in having UpdateEligibleNodes at this point? Can't this be derived from UpdateAddresses?

Reply:
i've gone back and forth on this a bit, but it's mostly because (1) i don't want gossip discovery polluting the trusted peers set and (2) i feel it's cleaner if connectivity manager is unaware of network address structure.